service-registry/src/main/java/io/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -143,12 +146,53 @@ private boolean registerSchemas() {
       boolean exists = srClient.isSchemaExist(microservice.getServiceId(), schemaId);
       LOGGER.info("schemaId {} exists {}", schemaId, exists);
       if (!exists) {
-        if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
-          return false;
+        if (serviceCenterEnvIsDev()) {
+            if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
+                return false;
+            }
+        }
+      } else {
+        //query if schema exist, if not exist, register schema&summary in SC
+        String serviceId = microservice.getServiceId();
+        String schema = srClient.getSchema(serviceId, schemaId);
+        if (null == schema) {
+          LOGGER.warn("schemacontent {}/{} does not exist.", serviceId, schemaId);
+          if (!srClient.registerSchema(serviceId, schemaId, content)) {
+            return false;
+          }
+        } else {
+          //query if schemaSummary exist, if not exist, register schema&summary in SC
+          String schemaSummary = srClient.getSchemaSummary(serviceId, schemaId);
+          if (null == schemaSummary) {
+            LOGGER.warn("schemasummary {}/{} does not exist.", serviceId, schemaId);
+            if (!srClient.registerSchema(serviceId, schemaId, content)) {
+              return false;
+            }
+          } else {
+        	//compare to localschema, if don't match,when SC runMode is dev,register schema&summary in SC
+            String localSummary = Hashing.sha256().newHasher().putString(content, StandardCharsets.UTF_8).hash().toString();
+            if (!localSummary.equals(schemaSummary)) {
+              LOGGER.warn("schemaSummary {}/{} is different from servicecenter.", serviceId, schemaId);
+              if (serviceCenterEnvIsDev()) {
+                if (!srClient.registerSchema(serviceId, schemaId, content)) {
+                  return false;
+                }
+              }
+            }
+          }
         }
       }
     }
 
     return true;
   }
+
+  private boolean serviceCenterEnvIsDev() {
+    if (Const.SERVICECENTER_RUNMODE_DEV.equals(srClient.getServiceCenterEnvironment().getRunMode())) {
```

Comment:
always query from remote?

Reply:
done, add a boolean scEnvIsDev to resolve